src/main/java/org/dsa/iot/calendar/Actions.java

```diff
@@ -321,50 +326,40 @@ public void handle(ActionResult event) {
+                        Date startDate = new SimpleDateFormat(BaseCalendar.DATE_PATTERN).parse(dates[0]);
+                        Date endDate = new SimpleDateFormat(BaseCalendar.DATE_PATTERN).parse(dates[1]);
+                        DSAEvent event = new DSAEvent(title, startDate.toInstant(), endDate.toInstant());
                         event.setDescription(desc);
-                        try {
-                            String[] dates = timeRange.split("/", 2);
-                            if (dates.length != 2) {
-                                throw new Exception("Unexpected dates length");
-                            }
-                            Date startDate = new SimpleDateFormat(BaseCalendar.DATE_PATTERN).parse(dates[0]);
-                            Date endDate = new SimpleDateFormat(BaseCalendar.DATE_PATTERN).parse(dates[1]);
-                            String location = actionResult.getParameter("location", new Value("")).getString();
-                            event.setStart(startDate);
-                            event.setEnd(endDate);
-                            event.setLocation(location);
-                            if (calendar.supportsMultipleCalendars()) {
-                                String calendar = actionResult.getParameter("calendar").getString();
-                                int indexOfPipe = calendar.lastIndexOf('|');
-                                String calTitle = calendar.substring(0, indexOfPipe);
-                                String calUid = calendar.substring(indexOfPipe + 1);
-                                event.setCalendar(new DSAIdentifier(calUid, calTitle));
-                            }
-                            calendar.createEvent(event);
-                            actionResult.getTable().addRow(Row.make(new Value("Event created.")));
-                        } catch (Exception e) {
-                            e.printStackTrace();
-                            actionResult.getTable().addRow(Row.make(new Value("Error occurred: " + e.getMessage())));
+                        String location = actionResult.getParameter("location", new Value("")).getString();
+                        event.setLocation(location);
+                        if (calendar.supportsMultipleCalendars()) {
+                            String calendar1 = actionResult.getParameter("calendar").getString();
+                            int indexOfPipe = calendar1.lastIndexOf('|');
+                            String calTitle = calendar1.substring(0, indexOfPipe);
+                            String calUid = calendar1.substring(indexOfPipe + 1);
+                            event.setCalendar(new DSAIdentifier(calUid, calTitle));
                         }
+                        calendar.createEvent(event);
+                        actionResult.getTable().addRow(Row.make(new Value("Event created.")));
+                    } catch (Exception e) {
+                        e.printStackTrace();
```

Comment:
Probably not good to print the stacktrace to stdio, need to use a logger or return error gracefully to the action.